projects/plugins/jetpack/class.jetpack.php

```diff
@@ -2435,19 +2447,36 @@ public static function get_available_modules( $min_version = false, $max_version
 				continue;
 			}
 
+			$mod_details = self::get_module( $slug );
+
+			if ( is_bool( $requires_connection ) && $requires_connection !== $mod_details['requires_connection'] ) {
+				continue;
+			}
+
+			if ( is_bool( $requires_user_connection ) && $requires_user_connection !== $mod_details['requires_user_connection'] ) {
+				continue;
+			}
+
 			$r[] = $slug;
 		}
 
 		return $r;
 	}
 
 	/**
-	 * Default modules loaded on activation.
+	 * Get default modules loaded on activation.
+	 *
+	 * @param bool|string $min_version Onlu return modules introduced in this version or later. Default is false, do not filter.
+	 * @param bool|string $max_version Only return modules introduced before this version. Default is false, do not filter.
+	 * @param bool|null   $requires_connection Only return modules that require a conncetion.
+	 * @param bool|null   $requires_user_connection Only return modules that require a user conncetion.
```

Comment:
Same as above.

Reply:
done in https://github.com/Automattic/jetpack/pull/19068/commits/c2a251af8671a72e356218d701cfd6884b3c88ee